consensus/src/experimental/commit_phase.rs

```diff
@@ -408,49 +407,38 @@ impl CommitPhase {
                                 unreachable!("Unexpected messages: something wrong with message dispatching.")
                             }
                         };
-                    }
-                    retry_cv = self.timeout_event_rx.select_next_some() => {
+                }
+                retry_cv = self.timeout_event_rx.select_next_some(), if self.blocks.is_some() => {
```

Comment:
I think we can deal with the drop issue by adding !self.commit_msg_rx.is_terminated()

Reply:
I'm still concerned whether "else" actually means "complete". In the document it says "else" will be executed if all other branches do not match. It might be just that the other branches are not ready. Let me do a quick local test.

Reply:
resolved